test.sh

```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+set -euo pipefail
```

Comment:
You could probably output the commands as they run. That's usually helpful. 
```suggestion
set -exuo pipefail
```